core/services/vrf/v2/integration_helpers_test.go

```diff
@@ -1736,3 +1737,145 @@ func testMaliciousConsumer(
 	}
 	require.Equal(t, 1, len(requests))
 }
+
+func testReplayOldRequestsOnStartUp(
+	t *testing.T,
+	ownerKey ethkey.KeyV2,
+	uni coordinatorV2UniverseCommon,
+	consumer *bind.TransactOpts,
+	consumerContract vrftesthelpers.VRFConsumerContract,
+	consumerContractAddress common.Address,
+	coordinator v22.CoordinatorV2_X,
+	coordinatorAddress common.Address,
+	batchCoordinatorAddress common.Address,
+	vrfOwnerAddress *common.Address,
+	vrfVersion vrfcommon.Version,
+	nativePayment bool,
+	assertions ...func(
+		t *testing.T,
+		coordinator v22.CoordinatorV2_X,
+		rwfe v22.RandomWordsFulfilled,
+		subID *big.Int),
+) {
+	sendingKey := cltest.MustGenerateRandomKey(t)
+	gasLanePriceWei := assets.GWei(10)
+	config, _ := heavyweight.FullTestDBV2(t, func(c *chainlink.Config, s *chainlink.Secrets) {
+		simulatedOverrides(t, assets.GWei(10), toml.KeySpecific{
+			// Gas lane.
+			Key:          ptr(sendingKey.EIP55Address),
+			GasEstimator: toml.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+		})(c, s)
+		c.EVM[0].MinIncomingConfirmations = ptr[uint32](2)
+		c.Feature.LogPoller = ptr(true)
+		c.EVM[0].LogPollInterval = models.MustNewDuration(1 * time.Second)
+	})
+	app := cltest.NewApplicationWithConfigV2AndKeyOnSimulatedBlockchain(t, config, uni.backend, ownerKey, sendingKey)
+
+	// Create a subscription and fund with 5 LINK.
+	subID := subscribeAndAssertSubscriptionCreatedEvent(t, consumerContract, consumer, consumerContractAddress, big.NewInt(5e18), coordinator, uni.backend, nativePayment)
+
+	// Fund gas lanes.
+	sendEth(t, ownerKey, uni.backend, sendingKey.Address, 10)
+	require.NoError(t, app.Start(testutils.Context(t)))
+
+	// Create VRF Key, register it to coordinator and export
+	vrfkey, err := app.GetKeyStore().VRF().Create()
+	require.NoError(t, err)
+	registerProvingKeyHelper(t, uni, coordinator, vrfkey)
+	keyHash := vrfkey.PublicKey.MustHash()
+
+	encodedVrfKey, err := app.GetKeyStore().VRF().Export(vrfkey.ID(), testutils.Password)
+	require.NoError(t, err)
+
+	// Shut down the node before making the randomness request
+	require.NoError(t, app.Stop())
+
+	// Make the first randomness request.
+	numWords := uint32(20)
+	requestID1, _ := requestRandomnessAndAssertRandomWordsRequestedEvent(t, consumerContract, consumer, keyHash, subID, numWords, 500_000, coordinator, uni.backend, nativePayment)
```

Comment:
So are we testing Scenario 1 described here: https://www.notion.so/chainlink/VRF-LogPoller-Operation-Guide-a0a62f9d76ec4e4e9d04959b87364a44

Reply:
yes correct